lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUBugException.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// Type of exception thrown when possible bugs are detected in IMRU code.
+    /// For example, we reach forbidden region of codes, inconsistent state etc.
+    /// </summary>
+    public sealed class IMRUBugException : Exception
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(IMRUBugException));
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="message">The user message for exception</param>
+        public IMRUBugException(string message)
+            : base(message)
+        {
+            Logger.Log(Level.Info, "Possible bug in the IMRU code");
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="message">The user message for exception</param>
+        /// <param name="inner">The actual exception message due to which connection failed</param>
+        public IMRUBugException(string message, Exception inner)
+            : base(message, inner)
+        {
+            Logger.Log(Level.Info, "Possible bug in the IMRU code");
+        }
+
+        /// <summary>
+        /// Appends the number of retries to the exception message.
+        /// </summary>
+        public override string Message
+        {
+            get { return "Possible Bug in the IMRU code: " + base.Message; }
```

Comment:
This seems a bit unintuitive. Perhaps you can define a `private static string` function that returns the `const string` appended to `message` and pass it directly to the `base` constructor.